src/main/java/net/coderbot/iris/shaderpack/ProgramSource.java

```diff
@@ -38,7 +44,7 @@ public ProgramSet getParent() {
 	}
 
 	public boolean isValid() {
-		return vertexSource != null && fragmentSource != null;
+		return vertexSource != null && geometrySource != null && fragmentSource != null;
```

Comment:
This logic is 100% broken. It's entirely valid for a program to lack a geometry shader, and most programs do in fact lack a geometry shader.

Reply:
what you're saying is it should be removed right?

Reply:
yes, since it is valid for a shader program to lack a geometry shader